pkg/virtual/framework/forwardingregistry/store.go

```diff
@@ -273,3 +290,29 @@ func clientGetter(dynamicClusterClient dynamic.ClusterInterface, namespaceScoped
 		}
 	}
 }
+
+// updateToCreateOptions creates a CreateOptions with the same field values as the provided PatchOptions.
+func updateToCreateOptions(uo *metav1.UpdateOptions) metav1.CreateOptions {
+	co := metav1.CreateOptions{
+		DryRun:          uo.DryRun,
+		FieldManager:    uo.FieldManager,
+		FieldValidation: uo.FieldValidation,
+	}
+	co.TypeMeta.SetGroupVersionKind(metav1.SchemeGroupVersion.WithKind("CreateOptions"))
+	return co
+}
+
+// hasUID return whether the provided object has a UID.
+func hasUID(obj runtime.Object) (bool, error) {
+	if obj == nil {
+		return false, nil
+	}
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
+		return false, apierrors.NewInternalError(err)
```

Comment:
this is bad style: better move the `NewInternalError` wrapping to the top-level, if it is not already there.

Reply:
done